bot/utils/functions.py

```diff
@@ -56,11 +68,26 @@ def parse_whitelist_message(
     )
 
 
-def format_segment(segment: json, episodes_per_season: int = 13) -> FormattedSegmentInfo:
+def format_segment(segment: json, season_info: Dict[str, int]) -> FormattedSegmentInfo:
     episode_info = segment.get("episode_info", {})
     total_episode_number = episode_info.get("episode_number", "Unknown")
-    season_number = (total_episode_number - 1) // episodes_per_season + 1 if isinstance(total_episode_number, int) else "Unknown"
-    episode_number_in_season = (total_episode_number - 1) % episodes_per_season + 1 if isinstance(total_episode_number, int) else "Unknown"
+
+    if not isinstance(total_episode_number, int):
+        return FormattedSegmentInfo(
+            episode_formatted="Unknown",
+            time_formatted="00:00",
+            episode_title=episode_info.get("title", "Unknown"),
+        )
+
+    season_number = 1
+    episodes_in_previous_seasons = 0
+    for season, episode_count in season_info.items():
+        if total_episode_number <= episodes_in_previous_seasons + episode_count:
```

Comment:
```suggestion
        if total_episode_number <= (episodes_in_previous_seasons + episode_count):
```

trace 50iq jak widze dzialania matematyczne w kodzie bez nawiasow XD